lighthouse-core/runner.js

```diff
@@ -86,31 +86,20 @@ class Runner {
         return artifacts;
       });
 
-      // Ignoring these two flags for coverage as this functionality is not exposed by the module.
-      /* istanbul ignore next */
-      if (opts.flags.saveArtifacts) {
-        run = run.then(artifacts => {
-          opts.flags.saveArtifacts && assetSaver.saveArtifacts(artifacts);
-          return artifacts;
-        });
-      }
-      if (opts.flags.saveAssets) {
-        run = run.then(artifacts => {
-          return assetSaver.saveAssets(opts, artifacts)
-            .then(_ => artifacts);
-        });
-      }
-
       // Run each audit sequentially, the auditResults array has all our fine work
       const auditResults = [];
       run = run.then(artifacts => config.audits.reduce((chain, audit) => {
         return chain.then(_ => {
           return Runner._runAudit(audit, artifacts);
         }).then(ret => auditResults.push(ret));
-      }, Promise.resolve()).then(_ => auditResults));
+      }, Promise.resolve()).then(_ => {
```

Comment:
I updated this to use "for .. of loop" and make those audit runs become a "standard run" (i.e. resolves to artifacts). Hopefully, this will increase readability.